kubeflow/trainer/rhai/traininghub_test.py

```
@@ -690,5 +690,127 @@ def test_instrumentation_wrapper_flush_all_prints():
+        ckpt_output_dir=str(ckpt_dir),
+        metrics_port=28080,
+    )
+
+    # Verify cleanup code is present
+    assert "Cleaning stale metrics files" in wrapper, "Should include cleanup logging"
+    assert "SFT_METRICS_FILE_PATTERN" in wrapper, "Should use SFT pattern"
+    assert "os.remove" in wrapper, "Should remove files"
+
+    # Execute the instrumentation code to test cleanup
+    exec_globals = {}
+    exec(wrapper, exec_globals)
```

Ruff (0.14.14)

[error] Use of `exec` detected

(S102)